src/verilog/verilog_synthesis.cpp

```diff
@@ -1400,8 +1452,32 @@ Function: verilog_synthesist::synth_decl
 \*******************************************************************/
 
 void verilog_synthesist::synth_decl(const verilog_declt &statement) {
+  // Look for supply0 and supply1 port class.
+  if(statement.get_class() == ID_supply0 || statement.get_class() == ID_supply1)
+  {
+    for(auto &op : statement.operands())
+    {
+      if(op.id() == ID_symbol)
+      {
+        const symbolt &symbol = ns.lookup(to_symbol_expr(op));
+
+        if(!symbol.is_state_var)
+        {
+          // much like a continuous assignment
+          const auto value =
+            make_supply_value(statement.get_class(), op.type());
+          verilog_continuous_assignt assign;
+          assign.add_source_location() = op.source_location();
+          assign.add_to_operands(equal_exprt(op, value));
```

Comment:
Looks like `verilog_continuous_assignt` is lacking a proper constructor?

Reply:
added one